src/api/sagaworker/devicedataworker.js

```diff
@@ -0,0 +1,20 @@
+import { call, put } from 'redux-saga/effects';
+import * as actions from '../../action/actions';
+import * as url from '../config';
+import fireApi from '../generic';
+
+export default function* submitDeviceData(action) {
+  try {
+    const response = yield call(fireApi, 'POST', url.deviceData, {
+      email_id: action.payload.email,
```

Comment:
this is just un-necessary code

email_id: action.payload.email,

if we structure payload so that it has email_id 

then we can directly post action.payload

so structure payload accordingly